src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -117,6 +117,19 @@ public void IncomingComplete(int count, Exception error)
             Complete();
         }
 
+        public void IncomingDeferred()
+        {
+            if (_pinned != null)
```

Comment:
Add a `Debug.Assert(_pinned != null)`. I'd really want to know if that wasn't the case.

You can keep the if check to be extra safe in prod (though there's an argument for just crashing).